swarm/src/handler.rs

```diff
@@ -224,6 +277,76 @@ pub trait ConnectionHandler: Send + 'static {
     {
         ConnectionHandlerSelect::new(self, other)
     }
+
+    /// Informs the handler about an event from the [`NetworkBehaviour`].
+    fn on_behaviour_event(&mut self, _event: Self::InEvent) {}
+
+    fn on_connection_event(
+        &mut self,
+        _event: ConnectionEvent<
+            Self::InboundProtocol,
+            Self::OutboundProtocol,
+            Self::InboundOpenInfo,
+            Self::OutboundOpenInfo,
+        >,
+    ) {
+    }
+}
```

Comment:
I wonder, once `Self::InEvent` is called `Self::FromBehaviour`, should we rename `ConnectionEvent` to `FromConnection`?

Reply:
Works for me, though I suggest not doing that within this pull request.